flink-connectors/flink-connector-kafka/src/test/java/org/apache/flink/connector/kafka/source/reader/KafkaPartitionSplitReaderTest.java

```diff
@@ -162,8 +161,18 @@ public void testNumBytesInCounter() throws Exception {
         assertThat(numBytesInCounter.getCount(), Matchers.greaterThan(latestNumBytesIn));
     }
 
-    @Test
-    public void testPendingRecordsGauge() throws Exception {
+    @ParameterizedTest
+    @EmptySource
+    @ValueSource(strings = {"_underscore.period-minus"})
+    public void testPendingRecordsGauge(String topicSuffix) throws Throwable {
+        final String topic1Name = TOPIC1 + topicSuffix;
+        final String topic2Name = TOPIC2 + topicSuffix;
+        if (!topicSuffix.isEmpty()) {
+            KafkaSourceTestEnv.setupTopic(
+                    topic1Name, true, true, KafkaSourceTestEnv::getRecordsForTopic);
+            KafkaSourceTestEnv.setupTopic(
+                    topic2Name, true, true, KafkaSourceTestEnv::getRecordsForTopic);
+        }
```

Comment:
Does it mean that the rest of the test works when `topicSuffix.isEmpty() == true` for `@EmptySource` and no topic has been created?

Reply:
Yes the test essentially setups two additional topics if the suffix is specified. `TOPIC1` and `TOPIC2` are already set up in the `setup` method.

Reply:
got it, thanks!